### PR TITLE
AWS OIDC: Script to configure EC2 Auto Discover with SSM

### PR DESCRIPTION
**Pull request by marcoandredinis**
There's a new teleport command that adds an inline policy which enables the AWS OIDC Integration to handle EC2 Auto Discover using the script mode (SSM).
It also adds the SSM Document used in the teleport installation.

There's also a new endpoint which returns the one off script that uses that command.

Demo:
```
$ teleport integration configure ec2-ssm-iam --role=MarcoTestRoleOIDCProvider --aws-region=us-east-1 --ssm-document-name=MarcoTestDoc001 --proxy-public-url=https://lenix.marcoandredinis.com:443
2024-04-30T16:42:50+01:00 INFO  IntegrationRole: IAM Policy added to Role policy:EC2DiscoverWithSSM role:MarcoTestRoleOIDCProvider awsoidc/ec2_ssm_iam_config.go:162
2024-04-30T16:42:50+01:00 INFO  SSM Document created name:MarcoTestDoc001 awsoidc/ec2_ssm_iam_config.go:180
$ teleport integration configure ec2-ssm-iam --role=MarcoTestRoleOIDCProvider --aws-region=us-east-1 --ssm-document-name=MarcoTestDoc001 --proxy-public-url=https://lenix.marcoandredinis.com:443
2024-04-30T16:42:54+01:00 INFO  IntegrationRole: IAM Policy added to Role policy:EC2DiscoverWithSSM role:MarcoTestRoleOIDCProvider awsoidc/ec2_ssm_iam_config.go:162
2024-04-30T16:42:55+01:00 INFO  SSM Document already exists name:MarcoTestDoc001 awsoidc/ec2_ssm_iam_config.go:173
```

### CODE REVIEW

**Comment by marcoandredinis** [comment]
@EdwardDowling @hugoShaka Can you please take a look?

File: lib/integrations/awsoidc/ec2_ssm_iam_config.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+	}
+
+	_, err = clt.PutRolePolicy(ctx, &iam.PutRolePolicyInput{
+		PolicyName:     &req.IntegrationRoleEC2SSMPolicy,
+		RoleName:       &req.IntegrationRole,
+		PolicyDocument: &ec2ICEPolicyDocument,
+	})
+	if err != nil {
+		if trace.IsNotFound(awslib.ConvertIAMv2Error(err)) {
+			return trace.NotFound("role %q not found.", req.IntegrationRole)
```
Comment (EdwardDowling) [inline lib/integrations/awsoidc/ec2_ssm_iam_config.go:157]:
```suggestion
			return trace.NotFound("role %q not found", req.IntegrationRole)
```

**Review by hugoShaka — APPROVED** [review]
LGTM, my only optional nit would be to have a flag or something so the `configure` command does not fail and force-update the SSM document. The user could just delete it, but having a `--force` or `--replace` would provide a simple upgrade path if we need, and make the command idempotent.

**Review by EdwardDowling — APPROVED** [review]

**Comment by marcoandredinis** [comment]
> LGTM, my only optional nit would be to have a flag or something so the `configure` command does not fail and force-update the SSM document. The user could just delete it, but having a `--force` or `--replace` would provide a simple upgrade path if we need, and make the command idempotent.

I think we can add that later if we ever change the script.

**Pull request merged**